libraries/AC_AutoTune/AC_AutoTune_Heli.cpp

```diff
@@ -1368,362 +1207,224 @@ void AC_AutoTune_Heli::set_gains_post_tune(AxisType test_axis)
         } else {
             // adjust tuning gain so max response gain is not exceeded
-            if (sp_prev_gain < max_resp_gain && gain[freq_cnt] > max_resp_gain) {
-                float adj_factor = (max_resp_gain - gain[freq_cnt]) / (gain[freq_cnt] - sp_prev_gain);
+            if (sp_prev_gain < max_resp_gain && test_data.gain > max_resp_gain) {
+                float adj_factor = (max_resp_gain - test_data.gain) / (test_data.gain - sp_prev_gain);
                 tune_p = tune_p + gain_incr * adj_factor; 
             }
             counter = AUTOTUNE_SUCCESS_COUNT;
         }
     }
     if (counter == AUTOTUNE_SUCCESS_COUNT) {
-        start_freq = 0.0f;  //initializes next test that uses dwell test
+        next_freq = 0.0f;  //initializes next test that uses dwell test
+        sweep_complete = false;
         reset_sweep_variables();
-        curr_test.freq = freq[0];
-        freq_cnt = 0;
-    } else {
-        start_freq = curr_test.freq;
-        stop_freq = curr_test.freq;
     }
 }
 
 // updating_max_gains: use dwells at increasing frequency to determine gain at which instability will occur
-void AC_AutoTune_Heli::updating_max_gains(float *freq, float *gain, float *phase, uint8_t &frq_cnt, max_gain_data &max_gain_p, max_gain_data &max_gain_d, float &tune_p, float &tune_d)
+void AC_AutoTune_Heli::updating_max_gains(sweep_info &test_data, float &next_freq, max_gain_data &max_gain_p, max_gain_data &max_gain_d, float &tune_p, float &tune_d)
 {
-    float test_freq_incr = 1.0f * M_PI * 2.0f;
-
-    if (!is_equal(start_freq,stop_freq)) {
-        frq_cnt = 2;
-        if (!is_zero(sweep.ph180.freq)) {
-            freq[frq_cnt] = sweep.ph180.freq - 0.5f * test_freq_incr;
-        } else {
-            freq[frq_cnt] = min_sweep_freq;
-        }
-        curr_test.freq = freq[frq_cnt];
-        start_freq = curr_test.freq;
-        stop_freq = curr_test.freq;
-
-    } else if (frq_cnt < 12 && is_equal(start_freq,stop_freq)) {
-        if (frq_cnt > 2 && phase[frq_cnt] > 161.0f && phase[frq_cnt] < 270.0f &&
-            !find_middle && !found_max_p) {
-            find_middle = true;
-        } else if (find_middle && !found_max_p) {
-            if (phase[frq_cnt] > 161.0f) {
-                // interpolate between frq_cnt-2 and frq_cnt
-                max_gain_p.freq = linear_interpolate(freq[frq_cnt-2],freq[frq_cnt],161.0f,phase[frq_cnt-2],phase[frq_cnt]);
-                max_gain_p.gain = linear_interpolate(gain[frq_cnt-2],gain[frq_cnt],161.0f,phase[frq_cnt-2],phase[frq_cnt]);
-            } else {
-                // interpolate between frq_cnt-1 and frq_cnt
-                max_gain_p.freq = linear_interpolate(freq[frq_cnt],freq[frq_cnt-1],161.0f,phase[frq_cnt],phase[frq_cnt-1]);
-                max_gain_p.gain = linear_interpolate(gain[frq_cnt],gain[frq_cnt-1],161.0f,phase[frq_cnt],phase[frq_cnt-1]);
-            }
-            max_gain_p.phase = 161.0f;
+    float test_freq_incr = 0.5f * M_2PI;
```

Comment:
```suggestion
    float test_freq_incr = M_PI;
```

Reply:
I do this to keep better track of what frequency I use in hz.  So i am going to keep it the way it is.